src/main/CommandLine.cpp

```diff
@@ -1041,30 +1041,109 @@ runReportLastHistoryCheckpoint(CommandLineArgs const& args)
         });
 }
 
+std::vector<std::pair<uint32_t, uint32_t>>
+parseSimulateSleepPerOp(std::vector<std::string> const& inputs)
+{
+    // The input must be of the form percentage/duration
+    // where
+    //     * 0 <= percentage <= 100
+    //     * 0 <= duration
+    //     * percentage for slow + percentage for medium + percentage for fast =
+    //     100.
+    //     * duration for slow >= duration for medium >= duration for fast.
+    if (std::any_of(inputs.begin(), inputs.end(),
+                    [](std::string const& s) { return s.empty(); }))
+    {
+        throw std::runtime_error(
+            "All three of simulate-sleep-per-op-{slow, medium, fast} must be "
+            "defined if at least one is defined");
+    }
+    auto parser = [](std::string const& input, uint32_t& percentage,
+                     uint32_t& duration) {
+        auto separatorIndex = input.find("/");
```

Comment:
do not roll your own parser: you have several bugs here (doesn't handle trailing characters for example).

Best is to use a regex in this kind of situation and only feed proper data to the integer conversion functions.

Reply:
Updated it with regex.